File: src/main/java/gg/mineral/practice/bukkit/events/PlayerDamageByPlayerEvent.java
package gg.mineral.practice.bukkit.events;

import lombok.Getter;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageEvent;

@Getter
public class PlayerDamageByPlayerEvent extends PlayerDamageEvent {

    private final Player damager;

    public PlayerDamageByPlayerEvent(Player damager, EntityDamageEvent entityDamageEvent) {
        super(entityDamageEvent);
        this.damager = damager;
    }

    public Player getActor() {
        return this.getDamager();
    }
}

File: src/main/kotlin/gg/mineral/practice/bukkit/events/PlayerDamageByPlayerEvent.kt
package gg.mineral.practice.bukkit.events

import org.bukkit.entity.Player
import org.bukkit.event.entity.EntityDamageEvent

class PlayerDamageByPlayerEvent(val damager: Player, entityDamageEvent: EntityDamageEvent) :
    PlayerDamageEvent(entityDamageEvent)
